spec/requests/messages_request_spec.rb

```diff
@@ -46,33 +62,57 @@
     end
   end
 
-  describe 'GET /request' do
+  describe 'GET /:organization_id/message/:message_id/request' do
     let(:user) { create(:user, organizations: [message.organization]) }
     let(:contributor) { create(:contributor, first_name: 'Zora', last_name: 'Zimmermann', organization: organization) }
+    let(:message) { create(:message, sender: contributor, recipient: nil, organization: organization) }
 
-    before(:each) { get(organization_message_request_url(message.organization, message, as: user)) }
-
-    context 'given an inbound message' do
-      let(:message) { create(:message, sender: contributor, recipient: nil, request: request) }
+    it_behaves_like 'unauthenticated' do
+      before { get(organization_message_request_url(message.organization, message)) }
+    end
 
-      it 'renders successfully' do
-        expect(response).to be_successful
-        expect(response.body).to include('Zora Zimmermann')
-      end
+    it_behaves_like 'protected' do
+      before { get(organization_message_request_url(message.organization, message, as: create(:user))) }
     end
 
-    context 'given an outbound message' do
-      let(:message) { create(:message, sender: nil, recipient: contributor, broadcasted: true) }
+    context 'with a permitted user' do
+      let(:user) { create(:user, organizations: [organization]) }
+      context 'with a message of that organization' do
+        before(:each) { get(organization_message_request_url(message.organization, message, as: user)) }
+
+        context 'given an inbound message' do
+          let(:message) { create(:message, sender: contributor, recipient: nil, organization: organization) }
+
+          it 'renders successfully' do
+            expect(response).to be_successful
+            expect(response.body).to include('Zora Zimmermann')
+          end
+        end
+
+        context 'given an outbound message' do
+          let(:message) { create(:message, sender: nil, recipient: contributor, broadcasted: true, organization: organization) }
```

Comment:
```suggestion
          let(:message) { create(:message, sender: user, recipient: contributor, broadcasted: true, organization: organization) }
```
Let's not add new tests with sender nil. While this is valid, for now, this comes from a time where we did not add the user as the sender and after some time, we will remove this possibility. Since the feature was introduced, we always add the current_user as the message sender.

Reply:
good to know. thnx.